client/src/client.rs

```diff
@@ -1054,6 +1050,7 @@ pub trait RpcApi: Sized {
 /// Client implements a JSON-RPC client for the Bitcoin Core daemon or compatible APIs.
 pub struct Client {
     client: jsonrpc::client::Client,
+    version: RwLock<Option<usize>>,
```

Comment:
maybe AtomicUsize?

Reply:
hmm Atomic with usize::max as default sounds better than `Once` and everything, especially because it's fine if we call `get_version()` more than once if there's a race.

But it seems like @shesek decided not to cache

Reply:
Yes, I should've used this instead of the RwLock. But I like @stevenroose's approach in #170 better.